src/SFA.DAS.EmployerIncentives.Web/Services/LegalEntities/LegalEntitiesService.cs

```diff
@@ -39,8 +43,15 @@ public async Task<IEnumerable<LegalEntityModel>> Get(string accountId)
 
         public async Task<LegalEntityModel> Get(string hashedAccountId, string hashedAccountLegalEntityId)
         {
-            var accountLegalEntityId = _hashingService.DecodeValue(hashedAccountLegalEntityId);
-            return await Get(hashedAccountId, accountLegalEntityId);
+            try
+            {
+                var accountLegalEntityId = _hashingService.DecodeValue(hashedAccountLegalEntityId);
+                return await Get(hashedAccountId, accountLegalEntityId);
+            }
+            catch (IndexOutOfRangeException) // hashed id contains invalid characters
+            {
+                return null;
```

Comment:
Should this return null or an empty object or just propagate the error?  The ValidateTermsSigned caller isn't checking if this is null so that controller would error instead.